File: br-server/app/api/routes/review.py
import uuid

from fastapi import APIRouter, Depends, HTTPException, Query, status
from sqlalchemy.ext.asyncio import AsyncSession

from app.api.dependencies import get_current_user_id, get_optional_current_user_id
from app.core.database import get_db
from app.schemas.review import (
    ReviewCreate,
    ReviewItem,
    ReviewListResponse,
    ReviewSummary,
)
from app.services import review_service
from app.services.review_service import DEFAULT_PAGE_SIZE, MAX_PAGE_SIZE

router = APIRouter(prefix="/api/v1/reviews", tags=["reviews"])


@router.get("", response_model=ReviewListResponse)
async def list_reviews(
    course_id: int | None = Query(None, description="按课程过滤"),
    teacher_id: int | None = Query(None, description="按老师过滤"),
    booking_id: int | None = Query(None, description="按订单过滤"),
    room_id: int | None = Query(None, description="按学习室过滤（仅自习座位订单）"),
    mine: bool = Query(False, description="只看本人发表的评价（含待审核/已驳回）"),
    rating_band: str = Query("all", pattern="^(all|good|mid|bad)$"),
    has_images: bool = Query(False, description="仅看有图"),
    sort: str = Query("new", pattern="^(new|score)$"),
    page: int = Query(1, ge=1),
    page_size: int = Query(DEFAULT_PAGE_SIZE, ge=1, le=MAX_PAGE_SIZE),
    db: AsyncSession = Depends(get_db),
    user_id: uuid.UUID | None = Depends(get_optional_current_user_id),
) -> ReviewListResponse:
    # 公开查询允许未登录访问，只有"我的评价"需要登录态
    if mine and user_id is None:
        raise HTTPException(status_code=status.HTTP_401_UNAUTHORIZED, detail="未登录")
    return await review_service.list_reviews(
        db,
        user_id=user_id,
        course_id=course_id,
        teacher_id=teacher_id,
        booking_id=booking_id,
        room_id=room_id,
        mine=mine,
        rating_band=rating_band,
        has_images=has_images,
        sort=sort,
        page=page,
        page_size=page_size,
    )


@router.get("/summary", response_model=ReviewSummary)
async def get_review_summary(
    course_id: int | None = Query(None),
    teacher_id: int | None = Query(None),
    room_id: int | None = Query(None),
    db: AsyncSession = Depends(get_db),
) -> ReviewSummary:
    if course_id is None and teacher_id is None and room_id is None:
        raise HTTPException(
            status_code=status.HTTP_422_UNPROCESSABLE_CONTENT,
            detail="course_id、teacher_id 与 room_id 至少传一个",
        )
    return await review_service.get_summary(
        db, course_id=course_id, teacher_id=teacher_id, room_id=room_id
    )


@router.post("", response_model=ReviewItem)
async def create_review(
    data: ReviewCreate,
    db: AsyncSession = Depends(get_db),
    user_id: uuid.UUID = Depends(get_current_user_id),
) -> ReviewItem:
    return await review_service.create_review(db, user_id, data)

File: br-app/src/api/review.js
import { get, post } from '@/utils/request'

/**
 * 获取学员评价列表
 * @param {Object} params - { course_id, teacher_id, room_id, booking_id, mine, rating_band, has_images, sort, page, page_size }
 *   rating_band: all | good(4-5星) | mid(3星) | bad(1-2星)
 *   sort: new(最新) | score(评分最高)
 *   room_id 按学习室过滤（仅自习座位订单，排除在该室上课的课程订单）
 *   mine 为 true 时返回本人全部状态（含待审核/已驳回），需登录态
 */
export function getReviewList(params) {
  return get('/api/v1/reviews', params)
}

/**
 * 获取评价概览（均分/总数/好评率/星级分布）
 * @param {Object} params - { course_id } 或 { teacher_id } 或 { room_id }，至少传一个
 */
export function getReviewSummary(params) {
  return get('/api/v1/reviews/summary', params)
}

/**
 * 发表评价
 * @param {Object} data - { booking_id, rating, content, images, tags, is_anonymous }
 */
export function createReview(data) {
  return post('/api/v1/reviews', data)
}
